### PR TITLE
Upgrade Docusaurus for new mobile UX

### PR DESCRIPTION
**Pull request by slorber**
## What does this PR do?

Handle feedbacks from @sw-yx.

Also updates the swizzled DocItem so that it renders the new mobile collapsible TOC

### CODE REVIEW

File: src/theme/DocItem/index.js [hunk 1]
```diff
@@ -73,6 +88,14 @@ function DocItem(props) {
                   </span>
                 </div>
               )}
+
+              {renderTocMobile && (
+                <TOCCollapsible
```
Comment (slorber) [inline src/theme/DocItem/index.js:93]: changes needed to render this

image

File: docusaurus.config.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ module.exports = {
             },
             {
               to: "/application-development",
-              activeBasePath: "/application-development",
+              activeBaseRegex: "(/application-development)|(/docs/(go|java|php|node))",
```
Comment (slorber) [inline docusaurus.config.js:47]: @sw-yx this permits to have the "SDKs" dropdown item to be highlighted when browsing any of the 4 SDKs. Docusaurus can't guess about those SDKs by itself, it just sees a regular page.

image


New mobile collapsible auto-expansion implemented in https://github.com/facebook/docusaurus/pull/5168:

image

(seen after navigating back to main sidebar menu)

**Review by slorber — COMMENTED** [review]
@sw-yx I think both items reported in your video feedback (https://github.com/facebook/docusaurus/pull/4273#issuecomment-877360318) have been fixed.

Let me know if you see anything weird left. 

If you see other issues regarding "active sidebar items", it's probably worth opening a new issue

File: docusaurus.config.js (hunk 2 above)
Comment (sw-yx) [reply docusaurus.config.js:47]: thats so smart! I never thought to use it like that.

**Review by sw-yx — APPROVED** [review]
this looks great!! thank you so much for the fix!

**Comment by slorber** [comment]
thanks for deploying, should help capture last edge cases 🤪 

Actually just found one, and you'll want to backport these changes in your swizzled DocItem: https://github.com/facebook/docusaurus/pull/5172

**Pull request merged**